Source/Charts/Charts/ChartViewBase.swift

```diff
@@ -51,43 +47,53 @@ open class ChartViewBase: NSUIView, ChartDataProvider, AnimatorDelegate
 {
     // MARK: - Properties
     
-    /// - Returns: The object representing all x-labels, this method can be used to
-    /// acquire the XAxis object and modify it (e.g. change the position of the
-    /// labels)
-    @objc open var xAxis: XAxis
-    {
-        return _xAxis
-    }
-    
     /// The default IValueFormatter that has been determined by the chart considering the provided minimum and maximum values.
-    internal var _defaultValueFormatter: IValueFormatter? = DefaultValueFormatter(decimals: 0)
-    
+    internal lazy var defaultValueFormatter: ValueFormatter = DefaultValueFormatter(decimals: 0)
+
     /// object that holds all data that was originally set for the chart, before it was modified or any filtering algorithms had been applied
-    internal var _data: ChartData?
-    
+    @objc open var data: ChartData?
+        {
+        didSet
+        {
+            offsetsCalculated = false
+
+            guard let data = data else { return }
+
+            // calculate how many digits are needed
+            setupDefaultFormatter(min: data.yMin, max: data.yMax)
+
+            for set in data.dataSets
+            {
+                if set.valueFormatter === defaultValueFormatter
```

Comment:
I'm kind of confused for this, as the old logic is 
```swift
// calculate how many digits are needed	
            setupDefaultFormatter(min: _data.getYMin(), max: _data.getYMax())	

            for set in _data.dataSets	
            {	
                if set.needsFormatter || set.valueFormatter === _defaultValueFormatter	
                {	
                    set.valueFormatter = _defaultValueFormatter	
                }	
            }	

            // let the chart know there is new data	
            notifyDataSetChanged()
```
which will hit set.needsFormatter so set.valueFormatter is set anyway. But the new code is confusing, as if it really `===`, there is no reason to set again?

besides, this is calling bar charts and pie charts unittests failing

I would like to remove it and try to modify the UT to avoid updating test images.

@jjatie @danielgindi what you think?